### PR TITLE
chore: Render handler package implementation

### PR DESCRIPTION
**Pull request by sebastian-barra-jalafund**
## What did I do?
I made an implementation to represent the render handler UML in code.

image

## How did I do it?
With Java language and based on the UML of the render handler.

## Why did I do it?
In order to have the render handler in code and start development.

### CODE REVIEW

File: bridge/src/main/java/com/bridge/renderHandlerImplementation/sound/SoundPlayer.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    public void addSound(Sound sound) {
+        sounds.add(sound);
+    }
+
+    public void removeSound(Sound sound) {
+        sounds.remove(sound);
+    }
+
+    public void removeSoundFromSprite(Sprite sprite) {
+
```
Comment (fabianromeroclaros) [inline bridge/src/main/java/com/bridge/renderHandlerImplementation/sound/SoundPlayer.java]: The Sound attribute of Sound should be removed

File: bridge/src/main/java/com/bridge/renderHandlerImplementation/sprite/Sprite.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package com.bridge.renderHandlerImplementation.sprite;
+
+public class Sprite {
```
Comment (fabianromeroclaros) [inline bridge/src/main/java/com/bridge/renderHandlerImplementation/sprite/Sprite.java]: All Sprite, Sound, Coord, IFile, etc already exists on the project, in the package piece, this have to be moved inside package render handler

**Review by fabianromeroclaros — CHANGES_REQUESTED** [review]
Also render method should be called in game loop

File: bridge/src/main/java/com/bridge/renderHandlerImplementation/sound/SoundPlayer.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    public void removeSound(Sound sound) {
+        sounds.remove(sound);
+    }
+
+    public void removeSoundFromSprite(Sprite sprite) {
+        
+    }
+
+    public void playSound(Sprite sprite) {
```
Comment (fabianromeroclaros) [inline bridge/src/main/java/com/bridge/renderHandlerImplementation/sound/SoundPlayer.java]: this method should be native method

File: bridge/src/main/java/com/bridge/renderHandlerImplementation/sound/SoundPlayer.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+       
+    }
+
+    public void playSound(Sound sound) {
```
Comment (fabianromeroclaros) [inline bridge/src/main/java/com/bridge/renderHandlerImplementation/sound/SoundPlayer.java]: this method should be native method

File: bridge/src/main/java/com/bridge/renderHandlerImplementation/sprite/SpriteRenderer.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    public void updateSprite(Sprite sprite, Sprite newSprite) {
+        int index = sprites.indexOf(sprite);
+        if (index != -1) {
+            sprites.set(index, newSprite);
+        }
+    }
+
+    public void renderSprite(Sprite sprite) {
```
Comment (fabianromeroclaros) [inline bridge/src/main/java/com/bridge/renderHandlerImplementation/sprite/SpriteRenderer.java]: this method should be native method

**Review by fabianromeroclaros — CHANGES_REQUESTED** [review]
And add documentation to all your implementation due this a contract for the developers.

**Comment by fabianromeroclaros** [comment]
The UML is this:
image

File: bridge/src/main/java/com/bridge/renderHandlerImplementation/sprite/Sprite.java (hunk 2 above)
Comment (sebastian-barra-jalafund) [reply bridge/src/main/java/com/bridge/renderHandlerImplementation/sprite/Sprite.java]: done

File: bridge/src/main/java/com/bridge/renderHandlerImplementation/sound/SoundPlayer.java (hunk 3 above)
Comment (sebastian-barra-jalafund) [reply bridge/src/main/java/com/bridge/renderHandlerImplementation/sound/SoundPlayer.java]: done

File: bridge/src/main/java/com/bridge/renderHandlerImplementation/sprite/SpriteRenderer.java (hunk 5 above)
Comment (sebastian-barra-jalafund) [reply bridge/src/main/java/com/bridge/renderHandlerImplementation/sprite/SpriteRenderer.java]: done

File: bridge/src/main/java/com/bridge/renderHandlerImplementation/sound/SoundPlayer.java (hunk 4 above)
Comment (sebastian-barra-jalafund) [reply bridge/src/main/java/com/bridge/renderHandlerImplementation/sound/SoundPlayer.java]: done

File: bridge/src/main/java/com/bridge/Game.java [hunk 6]
```diff
@@ -2,7 +2,11 @@
 
 import com.bridge.core.exceptions.GameException;
 import com.bridge.gamesettings.AGameSettings;
+import com.bridge.piece.render.RenderManager;
+import com.bridge.piece.sprite.SpriteFactory;
+import com.bridge.piece.sprite.Sprite;
 import com.bridge.processinputhandler.InputVerifier;
+//import com.bridge.renderHandlerImplementation.sprite.Sprite;
```
Comment (fabianromeroclaros) [inline bridge/src/main/java/com/bridge/Game.java]: Delete this comment please

File: bridge/src/test/java/com/bridge/RenderManager/RenderManagerTest.java
Comment (fabianromeroclaros) [inline bridge/src/test/java/com/bridge/RenderManager/RenderManagerTest.java:1]: Add also to the bridge/src/test/java/com/bridge/GameTest.java, a test related to the render manager in the loop please

**Review by fabianromeroclaros — CHANGES_REQUESTED** [review]

File: bridge/src/main/java/com/bridge/Game.java (hunk 6 above)
Comment (sebastian-barra-jalafund) [reply bridge/src/main/java/com/bridge/Game.java]: done

File: bridge/src/test/java/com/bridge/RenderManager/RenderManagerTest.java
Comment (sebastian-barra-jalafund) [reply bridge/src/test/java/com/bridge/RenderManager/RenderManagerTest.java:1]: done

**Review by fabianromeroclaros — CHANGES_REQUESTED** [review]
Change package "piece" name to "renderhandler"

File: bridge/src/test/java/com/bridge/RenderManager/RenderManagerTest.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
+package com.bridge.renderManager;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.bridge.piece.Coord;
+import com.bridge.piece.Size;
+import com.bridge.piece.render.RenderManager;
+import com.bridge.piece.sound.*;
```
Comment (fabianromeroclaros) [inline bridge/src/test/java/com/bridge/RenderManager/RenderManagerTest.java]: update the package import here as it has changed to "renderhandler"

File: bridge/src/main/java/com/bridge/renderHandler/sprite/SpriteRenderer.java [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+     * Renders a specific sprite.
+     *
+     * @param sprite the sprite to render
+     */
+    public native void renderSprite(Sprite sprite);
+
+    /**
+     * Renders all sprites in the list.
+     */
+    public void renderAllSprites() {
```
Comment (fabianromeroclaros) [inline bridge/src/main/java/com/bridge/renderHandler/sprite/SpriteRenderer.java]: Add a custom exception as in the other components

File: bridge/src/main/java/com/bridge/renderHandler/sound/SoundPlayer.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+     *
+     * @param sound the sound to play
+     */
+    public native void playSound(Sound sound);
+
+    /**
+     * Plays all sounds in the list.
+     */
+    public void playAllSounds() {
+        for (Sound sound : sounds) {
```
Comment (fabianromeroclaros) [inline bridge/src/main/java/com/bridge/renderHandler/sound/SoundPlayer.java:76]: Add a custom exception as in the other components

**Review by fabianromeroclaros — DISMISSED** [review]

File: bridge/src/test/java/com/bridge/RenderManager/RenderManagerTest.java (hunk 7 above)
Comment (sebastian-barra-jalafund) [reply bridge/src/test/java/com/bridge/RenderManager/RenderManagerTest.java]: name updated

File: bridge/src/main/java/com/bridge/renderHandler/sound/SoundPlayer.java (hunk 9 above)
Comment (sebastian-barra-jalafund) [reply bridge/src/main/java/com/bridge/renderHandler/sound/SoundPlayer.java:76]: custom exception added

File: bridge/src/main/java/com/bridge/renderHandler/sprite/SpriteRenderer.java (hunk 8 above)
Comment (sebastian-barra-jalafund) [reply bridge/src/main/java/com/bridge/renderHandler/sprite/SpriteRenderer.java]: custom exception added

**Review by fabianromeroclaros — APPROVED** [review]

**Review by EmanuelGalindoC — APPROVED** [review]

**Pull request merged**